neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/utils/BoltNeo4jUtils.java

```diff
@@ -117,4 +121,30 @@ public static void closeSafely(AutoCloseable closeable, Logger logger) {
         }
     }
 
+    public static boolean hasResultSet(BoltNeo4jConnection connection, String cypher) {
+        try (final Session session = connection.newNeo4jSession();
+             final Transaction transaction = session.beginTransaction()) {
+            if (StringUtils.isBlank(cypher)) {
```

Comment:
it's probably better to perform this check before opening a new session or transaction, i.e. move it to the top of the function

Reply:
done